recipes-openxt/qemu-dm/qemu-dm-2.6.2/acpi-pm-feature.patch

```diff
@@ -1029,17 +1035,79 @@ Index: qemu-2.6.2/hw/xen/xen_acpi_pm.c
 +    memory_region_add_subregion(s->space_io, ACPI_PM_STATUS_PORT, &s->mr);
 +}
 +
++/* -------/ Power Button Status /------------------------------------------- */
++
++/*
++ * Report SLP/WAK bits for power and sleep buttons.
++ *
++ * Returns bits set
++ */
++static uint64_t acpi_button_sts_read(void *opaque, hwaddr addr, uint32_t size)
++{
++    XenACPIPMState *s = opaque;
++    if (addr == 0) {
++        return s->button_status;
++    }
++
++    return 0xff;
++}
++
++/*
++ * Track state of SLP/WAK bits for power and sleep buttons.  The fields
++ * are WriteAsZero, writing 1 clears the bit.
++ */
++static void acpi_button_sts_write(void * opaque, hwaddr addr,
++                                  uint64_t val, uint32_t size)
++{
++    XenACPIPMState *s = opaque;
++
++    if (addr == 0) {
++        s->button_status &= ~val;
```

Comment:
So button_status is only ever set by other QEMU code for the the two bits.  The above "write" is really only clearing set bits, so we can never set additional bits.

I can mask down val, but I didn't think it was necessary.